LibTest/js_interop_unsafe/JSObjectUnsafeUtilExtension/index_operator_A01_t01.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion JSAny? operator []( String property )
+/// Shorthand helper for [getProperty] to get the value of the property key
+/// `property` of this [JSObject], but takes a Dart value.
+///
+/// @description Check that this operator returns the value of the property key
+/// `property` of this [JSObject].
+/// @author sgrekhov22@gmail.com
+
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+import '../../../Utils/expect.dart';
+import '../../js_interop/js_utils.dart';
+
+main() {
+  eval(r'''
+    globalThis.object1 = {
+      p1: 1,
+      p2: "2",
+      p3: function() {}
+    };
+    
+    function Foo() {
+      this.p1 = 1;
+      this.p2 = "2";
+      this.p3 = function() {};
+    };
+    globalThis.object2 = new Foo();
+    
+    class Bar {
+      constructor() {
+        this.p1 = 1;
+        this.p2 = "2";
+        this.p3 = function() {};
+      }
+    };
+    globalThis.object3 = new Bar();
+  ''');
+  test(globalContext["object1"] as JSObject);
+  test(globalContext["object2"] as JSObject);
+  test(globalContext["object3"] as JSObject);
+}
+
+test(JSObject obj) {
+  Expect.equals(1, obj["p1"].dartify());
```

Comment:
What's the trade-off when comparing `.dartify()` and `toDartInt` here? I'd expect that `toDartInt` is preferred when it is known to work, but it could of course be claimed that returning any kind of Dart object and failing in `Expect.equals` is nicer than throwing in `toDartInt` because it wasn't an `int` in the first place.